lib/linux_stats/os/proc_mounts.rb

```diff
@@ -42,23 +48,50 @@ module DataFile
   class Reporter
 
     attr_accessor :blocks_per_kilobyte,
-                  :mounted_partitions
+                  :mounted_partitions,
+                  :container_prefix
 
-    def initialize(data_directory = PROC_DIRECTORY_DEFAULT)
+    def initialize(data_directory = PROC_DIRECTORY_DEFAULT, container_mount_name =
+        CONTAINER_MOUNT_PREFIX, data = nil, test_mode = false)
       set_data_paths data_directory
+      set_container_mount container_mount_name
       @blocks_per_kilobyte = 4 # TODO: calculate from info in /proc?  Where?
-      @mounted_partitions = mounts
+      @test_mode = test_mode
+      if data && @test_mode
+        @mounted_partitions = mounts(data)
+      else
+        @mounted_partitions = mounts
+      end
     end
 
     def set_data_paths(data_directory = nil)
       @proc_data_source = "#{data_directory}#{DataFile::MOUNTS_PATH}"
     end
 
+    def set_container_mount(container_mount_name = nil)
+      @container_prefix = container_mount_name
+    end
+
     def report
       # execution time: 0.3 ms  [LOW]
       storage_report = {}
+
       mounted_partitions.each do |partition|
+        # Because of the way this runs on successive iterations (via the binary or within a Ruby
+        # process using this as a library), we need to manually re-build the host path when
+        # operating against data using when he are operating in the defined container mode.
+        if (@proc_data_source.include? 'hostproc') && (!partition.include? @container_prefix)
```

Comment:
should this be 
if (@proc_data_source.include? '/hostproc')
In order to handle the case where there might be another mount called hostproc? I know its a longshot, but...